checkpoint_schedules/none_revolver.py

```diff
@@ -73,11 +93,103 @@ def uses_storage_type(self, storage_type):
         """
 
         return storage_type == self._storage
+    
+
+class SingleDiskStorageSchedule(CheckpointSchedule):
+    """A checkpointing schedule where all adjoint dependencies
+    are stored in disk.
+
+    Notes
+    -----
+    Online, unlimited adjoint calculations permitted.
+
+    Parameters
+    ----------
+    storage : enum
+        Indicate that the execution should stores in memore all foward restart 
+        data and non-linear dependency data.
+
+    Notes
+    -----
+    Online, unlimited adjoint calculations permitted.
+
+    `write_ics` is always `False` for this schedule by considering that storing
+    the forward restart data is unnecessary by this schedule, as there is no
+    need to recompute the forward solver while time advancing the adjoint
+    solver.
+    """
+
+    def __init__(self, move_data=False):
+        self._move_data = move_data
+        self._storage = StorageType.ADJ_DEPS
+        super().__init__()
+    
+    def _iterator(self):
+        """Schedule iterator.
+        """
+
+        if self._max_n is not None:
+            # Unexpected finalize
+            raise RuntimeError("Invalid checkpointing state")
+
+        while self._max_n is None:
+            n0 = self._n
+            n1 = n0 + sys.maxsize
+            self._n = n1
+            yield Forward(n0, n1, False, True, StorageType.DISK)
+
+        yield EndForward()
+
+        for i in range(self._max_n, 0, -1):
+            if self._r < self._max_n:
+                # Reverse
+                if self._move_data is True:
```

Comment:
`if self._move_data`. It's unsafe to test using `is True` or `is False`, e.g. `np.int64(1) == 1 is True` is `False`.

Reply:
Yes. 🤦🏻‍♀️

Reply:
Yes. 🤦🏻‍♀️